### PR TITLE
Issue #7564: Add default config examples for ConstantName

### PR DESCRIPTION
**Pull request by RhnSaxena**
Fixes Issue: #7564 Updated the ConstantName with examples for default configuration.

### CODE REVIEW

**Comment by rnveach** [comment]
@RhnSaxena Please see https://github.com/checkstyle/checkstyle/wiki/PR-rules.
Speifically items 1, 3, and 4.

**Comment by rnveach** [comment]
@romani ping, we have a wercker failure in master

**Comment by romani** [comment]
wercker failue:
`[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-common-rules) @ hibernate-search-parent ---
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.5.2 is not in the allowed range 3.6.2.`

we need to bump version of maven in wercker, I will send PR to fix this.

**Comment by romani** [comment]
@RhnSaxena , wercker is fixed, please rebase on latest master code to make wercker green

**Comment by rnveach** [comment]
@RhnSaxena Your PR should only have 1 commit in it, which should just be your changes for the check.  You currently have 4 commits in this PR.

**Comment by RhnSaxena** [comment]
@romani wercker is green now. 
@rnveach so sorry about that, will make sure to do it in 1 commit from now on.

**Comment by romani** [comment]
@RhnSaxena , all CI should be green.
Please review https://travis-ci.org/checkstyle/checkstyle/jobs/654044432#L1353
Before you push to remote make sure `mvn verify` is passed on your local.

File: src/it/it.iml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/it/it.iml]: from CI:
> `Resource must start with 'Input' or 'Expected': com/puppycrawl/tools/checkstyle/checks/imports/imports.iml ==> expected: <true> but was: <false>`

please remove this file, it is not allowed to have files other than java files.

File: src/it/resources-noncompilable/resources-noncompilable.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/it/resources-noncompilable/resources-noncompilable.iml]: please remove this file, some IDE creates it, we do not need it.

File: src/test/test.iml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
```
Comment (romani) [inline src/test/test.iml]: remove this file

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/sizes/sizes.iml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/sizes/sizes.iml]: remove this file

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/imports.iml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/imports.iml]: remove this file

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/coding.iml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/coding.iml]: remove this file.

File: src/main/main.iml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
```
Comment (romani) [inline src/main/main.iml]: remove this file

**Comment by RhnSaxena** [comment]
@romani @rnveach ping, all ci tests are green now. Please review.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 8]
```diff
@@ -59,7 +59,38 @@
  * <pre>
  * &lt;module name="ConstantName"/&gt;
  * </pre>
- *
+ * <p>
+ *   The following configuration is applicable by default:
+ * </p>
+ * <pre>
+ * &lt;module name="ConstantName"&gt;
+ * &lt;property name="format" value="^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/&gt;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: Indentation is lost in javadoc that xdoc has. They should be as similar as possible.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 9]
```diff
@@ -59,7 +59,38 @@
  * <pre>
  * &lt;module name="ConstantName"/&gt;
  * </pre>
- *
+ * <p>
+ *   The following configuration is applicable by default:
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: Too much indentation here.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 10]
```diff
@@ -73,16 +104,15 @@
...
- *   final static int logger = 50; // OK
- *   final static int logMYSELF = 10; // violation, name 'logMYSELF' must match
- *                                    // pattern '^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
- *   final static int loggerMYSELF = 5; // violation, name 'loggerMYSELF' must match
- *                                      // pattern '^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
- *   final static int MYSELF = 100; // OK
- *   final static int myselfConstant = 1; // violation, name 'myselfConstant' must match pattern
- *                                        // '^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
- * }
+ * final static int log = 10; // OK
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: These lines don't look like they should have been changed. All I see is indentation being removed and no reason why it should have.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 11]
```diff
@@ -97,13 +127,12 @@
...
  * <pre>
  * class MyClass {
- *   public final static int firstConstant = 10; // OK
- *   protected final static int secondConstant = 100; // OK
- *   final static int thirdConstant = 1000; // violation, name 'thirdConstant' must
- *                                          // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
- *   private final static int fourthConstant = 50; // violation, name 'fourthConstant' must match
- *                                                 // pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
- * }
+ * public final static int firstConstant = 10; // OK
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: same.

**Comment by rnveach** [comment]
If you need to make fixes to the PR, do not close the PR. You will need to amend your current branch and force push the changes to github.
https://github.com/checkstyle/checkstyle/wiki/PR-rules
Item number 9.

**Comment by RhnSaxena** [comment]
@rnveach I only removed the commits, they were more than one. After removing them the PR closed automatically. This time I will make the changes in one commit and take into consideration all the rules.

**Comment by RhnSaxena** [comment]
Fixes Issue: #7564 
Add default config example for ConstantName

**Comment by RhnSaxena** [comment]
@rnveach please review.

File: src/it/it.iml (hunk 1 above)
Comment (romani) [reply src/it/it.iml]: @RhnSaxena, please do not resolve conversation. Author will do this, it is reminder to recheck all points for author.

**Comment by romani** [comment]
Please update PR description as requested at https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

File: src/xdocs/config_naming.xml [hunk 12]
```diff
@@ -658,6 +658,41 @@ class MyClass3&lt;abc&gt; {} // violation, the class type parameter
         <source>
 &lt;module name="ConstantName"/&gt;
         </source>
```
Comment (romani) [inline src/xdocs/config_naming.xml]: Please add code example for default config

**Comment by rnveach** [comment]
@RhnSaxena ping

**Comment by RhnSaxena** [comment]
> @RhnSaxena ping

@rnveach ping.

**Comment by baratali** [comment]
@RhnSaxena , travis-ci fails:
```
The PR is based on a master that is 29 commit(s) old.
The max allowed is 10.
This PR is too old and should be rebased on origin/master.
```
Please rebase your branch on master branch from checkstyle/checkstyle repo.

Also, please reply to each review note above with "Done" if it was done.

File: src/xdocs/config_naming.xml [hunk 13]
```diff
@@ -658,6 +658,41 @@ class MyClass3&lt;abc&gt; {} // violation, the class type parameter
...
+        <p>
+          The following configuration is applicable by default:
```
[the remark this answers is not in the document]
Comment (RhnSaxena) [reply src/xdocs/config_naming.xml]: Done

File: src/xdocs/config_naming.xml (hunk 12 above)
Comment (RhnSaxena) [reply src/xdocs/config_naming.xml]: Done

**Comment by RhnSaxena** [comment]
@baratali done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 14]
```diff
@@ -61,6 +61,26 @@
...
+ *                                  // '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   public final static int FIRST = 100; // OK
+ *   public final static int second = 1; // violation, name 'second' must match pattern
+ *                                       // '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   protected final static int THIRD = 100; // OK
+ *   protected final static int fOUR = 1; // violation, name 'fOUR' must
+ *                                        // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   private final static int FIVE = 100; // OK
+ *   private final static int Six = 1; // violation, name 'Six' must match pattern
+ *                                     // '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$' }
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: Please add a class field to show that it checks only constants

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 15]
```diff
@@ -61,6 +61,26 @@
...
+ *   final static int MYSELF = 100; // OK
+ *   final static int yourself = 1; // violation, name 'yourself' must match pattern
+ *                                  // '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   public final static int FIRST = 100; // OK
+ *   public final static int second = 1; // violation, name 'second' must match pattern
+ *                                       // '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   protected final static int THIRD = 100; // OK
+ *   protected final static int fOUR = 1; // violation, name 'fOUR' must
+ *                                        // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   private final static int FIVE = 100; // OK
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: Please rename one of the constants using underscore and numbers as well. To show that default pattern allows such names

**Comment by baratali** [comment]
@RhnSaxena , please add a screenshot of your 'Examples' section from documentation page.
See this issue's description https://github.com/checkstyle/checkstyle/pull/7703

**Comment by RhnSaxena** [comment]
@baratali I don't know how to view the documentation page. Can you please tell me how to do it?

**Comment by wilcoln** [comment]
@RhnSaxena, you can generate by running `mvn -e site` 
It will be located inside the `checkstyle/target/site` folder.
To view it, open any of the page of that folder with a browser.  
For this particular issue I believe you'll want to open the `config_coding.html` page.

**Comment by RhnSaxena** [comment]
ConstantName
@baratali this is the screenshot. please do review.

**Comment by romani** [comment]
@RhnSaxena , please update PR description as requested in issue. We will not review PR till all details provided. If something is still not clear, please review PRs that already merged for simialr update.

**Pull request closed without merging**